### PR TITLE
[TTAHUB-3095] Don't remove expired grants from the recipients dropdown when an AR is unlocked

### PR DESCRIPTION
**Pull request by nvms**
## Description of change

Modifies `possibleRecipients` to accept an optional report id. If the report id is provided, it returns recipients that *may be expired, but have been used on this report previously*.

It also adds a new endpoint

`/api/activity-report/activity-recipients/:reportId`

## How to test

Use the seeded data. Be Hermione. Give yourself unlock permissions. The report with id 9999 already has a recipient that is Inactive. Unlock report 9999. Edit it, go to the account summary, remove the recipient from the dropdown, expand the dropdown and you should still have the old recipient in the list even though they are expired. Save the report. Be sure the save takes effect.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3095

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by GarrettEHill — APPROVED** [review]
Looks fine, being we worked on it together

File: src/routes/activityReports/index.js [hunk 1]
```diff
@@ -40,6 +41,7 @@ router.post('/', transactionWrapper(createReport));
 router.get('/approvers', transactionWrapper(getApprovers));
 router.get('/groups', transactionWrapper(getGroups));
 router.get('/activity-recipients', transactionWrapper(getActivityRecipients));
+router.get('/activity-recipients/:reportId', transactionWrapper(getActivityRecipientsForExistingReport));
```
Comment (thewatermethod) [inline src/routes/activityReports/index.js:44]: might be a useless change, but how do we feel about flipping this to be``` /:reportId/activity-recipients``` (i.e ```/api/activity-reports/1/activity-recipients```)

File: src/routes/activityReports/handlers.js [hunk 2]
```diff
@@ -722,6 +722,14 @@ export async function getActivityRecipients(req, res) {
   res.json(activityRecipients);
 }
 
+export async function getActivityRecipientsForExistingReport(req, res) {
```
Comment (thewatermethod) [inline src/routes/activityReports/handlers.js:725]: Do we need to check permissions here?

**Comment by thewatermethod** [comment]
Everything looks great, just a question and a suggestion

File: src/routes/activityReports/index.js (hunk 1 above)
Comment (nvms) [reply src/routes/activityReports/index.js:44]: Yeah.. I had this same thought. I'll make this change 👍

File: src/routes/activityReports/handlers.js (hunk 2 above)
Comment (nvms) [reply src/routes/activityReports/handlers.js:725]: I'm not actually sure. What do you think? What we're potentially exposing (to already-authenticated users) is which grant recipients were previously associated with a given AR. My initial thought is that this isn't sensitive - but maybe I"m wrong

Comment (thewatermethod) [reply src/routes/activityReports/handlers.js:725]: I suppose without context that info doesn't feel particularly sensitive. The only counterargument is that all the other report-related data is confined to a region. I'm ok with site-access level permissions here, personally. Although, I am not the decider.

Comment (nvms) [reply src/routes/activityReports/handlers.js:725]: I ended up adding AR read permissions to this handler: 440a55a

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**